services/provider/server/consumer.go

```diff
@@ -211,24 +218,44 @@ func (c *ocsConsumerManager) UpdateStatusLastHeatbeat(ctx context.Context, id st
 	}
 	c.mutex.RUnlock()
 
-	patchInfo := struct {
-		Op    string      `json:"op"`
-		Path  string      `json:"path"`
-		Value interface{} `json:"value"`
-	}{
-		Op:    "replace",
-		Path:  "/status/lastHeartbeat",
-		Value: metav1.Now(),
+	patchInfo := []statusPatch{
+		{
+			Op:    "replace",
+			Path:  "/status/lastHeartbeat",
+			Value: metav1.Now(),
+		},
+	}
+
+	// TODO: fields doesn't exist in current iteration but exists in previous
```

Comment:
Can you pls explain this TODO. I am not able to understand it.

Reply:
Why it will be empty in the later state, unless someone reverts to the previous version of the client from the later one.

Reply:
- ya, now that I rethink about it, we are validating the data sent from client and even if one of the data point fails the status update will not happen, what I remember faintly was adding this comment when I didn't have version validation 🤔
- I'll remove this when I go over this code and couldn't justify it

thanks.

Reply:
is the info param to the function optional or the data inside the struct be optional?

Reply:
done